iOS_SDK/OneSignalSDK/Source/OSStateSynchronizer.m

```diff
@@ -81,27 +81,29 @@ - (OSUserStateSynchronizer *)getEmailStateSynchronizer {
         return nil;
 }
 
-- (void)registerUserWithState:(OSUserState *)registrationState withSuccess:(OSMultipleSuccessBlock)successBlock onFailure:(OSMultipleFailureBlock)failureBlock {
+- (NSArray<OSUserStateSynchronizer *> * _Nonnull)getStateSynchronizers {
+    NSMutableArray *stateSynchronizers = [NSMutableArray new];
+    
     let pushStateSyncronizer = [self getPushStateSynchronizer];
     let emailStateSyncronizer = [self getEmailStateSynchronizer];
+    [stateSynchronizers addObject:pushStateSyncronizer];
     
-    let pushDataDic = (NSMutableDictionary *)[registrationState.toDictionary mutableCopy];
-    pushDataDic[@"identifier"] = _currentSubscriptionState.pushToken;
+    if (emailStateSyncronizer)
+        [stateSynchronizers addObject:emailStateSyncronizer];
     
+    return stateSynchronizers;
+}
+
+- (void)registerUserWithState:(OSUserState *)registrationState withSuccess:(OSMultipleSuccessBlock)successBlock onFailure:(OSMultipleFailureBlock)failureBlock {
+    let stateSyncronizer = [self getStateSynchronizers];
+    // Begin constructing the request for the external id update
```

Comment:
This comment isn't correct, this is a register request. Recommend removing instead of updating comment as the method name already describes the request about to be made.